app/utility/expression_utilities.py

```diff
@@ -55,7 +61,7 @@ def _print_log(self, expr, exp=None):
 
 greek_letters = [
     "Alpha", "alpha", "Beta", "beta", "Gamma", "gamma", "Delta", "delta", "Epsilon", "epsilon", "Zeta", "zeta",
-    "Eta", "eta", "Theta", "theta", "Iota", "iota", "Kappa", "kappa", "Lambda", # "lambda" removed to avoid collision with reserved keyword in python
+    "Eta", "eta", "Theta", "theta", "Iota", "iota", "Kappa", "kappa", "Lambda",  # "lambda" removed to avoid collision with reserved keyword in python
```

Comment:
lower and upper-case symbols can have different meaning and can appear in the same expressions, and mixed/case (e.g. "eTa") is either meaningless or should be interpreted as implicit multiplication rather than a single symbol. I think using case-insensitive comparison would case more problems than it would solve here.